### PR TITLE
fix(aws): Assume role for Gov Cloud

### LINKED ISSUE #4244: [Bug]: Assume Role Region is not valid for GovCloud Regions
### Steps to Reproduce

1.  prowler aws --region us-gov-east-1 --role arn:aws-us-gov:iam::<account-id>:role/<role> --log-level DEBUG
2. aws
3. kubernetes
4. An error occurred (InvalidClientTokenId) when calling the AssumeRole operation: The security token included in the request is invalid

### Expected behavior

Assume role works properly with an applicable govcloud region. This is occurring because the AWS region being set does not use the region being set as a CLI argument.

### Actual Result with Screenshots or Logs

{"timestamp": "2024-06-11 20:04:10,218", "filename": "aws_provider.py:872", "level": "CRITICAL", "module": "aws_provider", "message": "ClientError[855] -- An error occurred (InvalidClientTokenId) when calling the AssumeRole operation: The security token included in the request is invalid"}

### How did you install Prowler?

From pip package (pip install prowler)

### Environment Resource

EKS

### OS used

Linux

### Prowler version

Prowler 4.2.4

### Pip version

pip 23.0.1

### Context

_No response_

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context

Fixes #4244 

### Description

Handle AWS Gov Cloud partition while assuming roles.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/aws_provider.py [hunk 1]
```diff
@@ -850,8 +850,7 @@ def assume_role(
                 mfa_info = self.__input_role_mfa_token_and_code__()
                 assume_role_arguments["SerialNumber"] = mfa_info.arn
                 assume_role_arguments["TokenCode"] = mfa_info.totp
-
-            sts_client = create_sts_session(session, AWS_STS_GLOBAL_ENDPOINT_REGION)
```
Comment (jfagoagas) [inline prowler/providers/aws/aws_provider.py:854]: This is the issue here, we were always assuming a role with the Global STS Endpoint Region which is `us-east-1`.

File: prowler/providers/aws/aws_provider.py [hunk 2]
```diff
@@ -424,7 +421,7 @@ def set_identity(
         logger.info(f"Original AWS Caller Identity ARN: {caller_identity.arn}")
 
         partition = parse_iam_credentials_arn(caller_identity.arn.arn).partition
-
+        print(partition)
```
Comment (MrCloudSec) [inline prowler/providers/aws/aws_provider.py]:
```suggestion
```

**Review by MrCloudSec — APPROVED** [review]
Thanks for the fix!!

**Comment by jfagoagas** [comment]
I'm not sure about this backport, we have to check it manually in v3.

**Pull request merged**